### ISSUE TITLE
[Bug]: unable to refund amount for an order with captured payment

### ISSUE THREAD

**Issue by usamaadev** [medusajs/medusa#10392]
### Package.json file

```JSON
{
  "name": "medusa-starter-default",
  "version": "0.0.1",
  "description": "A starter for Medusa projects.",
  "author": "Medusa (https://medusajs.com)",
  "license": "MIT",
  "keywords": [
    "sqlite",
    "postgres",
    "typescript",
    "ecommerce",
    "headless",
    "medusa"
  ],
  "scripts": {
    "build": "medusa build",
    "seed": "medusa exec ./src/scripts/seed.ts",
    "start": "medusa start",
    "dev": "medusa develop",
    "test:integration:http": "TEST_TYPE=integration:http NODE_OPTIONS=--experimental-vm-modules jest --silent=false --runInBand --forceExit",
    "test:integration:modules": "TEST_TYPE=integration:modules NODE_OPTIONS=--experimental-vm-modules jest --silent --runInBand --forceExit",
    "test:unit": "TEST_TYPE=unit NODE_OPTIONS=--experimental-vm-modules jest --silent --runInBand --forceExit"
  },
  "dependencies": {
    "@medusajs/admin-sdk": "latest",
    "@medusajs/cli": "latest",
    "@medusajs/framework": "latest",
    "@medusajs/medusa": "latest",
    "@mikro-orm/core": "5.9.7",
    "@mikro-orm/knex": "5.9.7",
    "@mikro-orm/migrations": "5.9.7",
    "@mikro-orm/postgresql": "5.9.7",
    "awilix": "^8.0.1",
    "medusa-fulfillment-manual": "^1.1.41",
    "pg": "^8.13.0"
  },
  "devDependencies": {
    "@medusajs/test-utils": "late0st",
    "@mikro-orm/cli": "5.9.7",
    "@swc/core": "1.5.7",
    "@swc/jest": "^0.2.36",
    "@types/jest": "^29.5.13",
    "@types/node": "^20.0.0",
    "@types/react": "^18.3.2",
    "@types/react-dom": "^18.2.25",
    "jest": "^29.7.0",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "ts-node": "^10.9.2",
    "typescript": "^5.6.2",
    "vite": "^5.2.11"
  },
  "engines": {
    "node": ">=20"
  }
}
```

### Node.js version

v20.9.0

### Database and its version

PostgreSQL 17.0

### Operating system name and version

Ubuntu 22.04.4 LTS

### What happended?

When attempting to process a refund on an order with a captured payment, the system throws the following error:

`Order does not have an outstanding balance to refund
`

Steps to Reproduce:

1. Place an order and capture the payment successfully.
2. Attempt to issue a refund for the captured payment.
3. Observe the error.

### Expected behavior

The refund should process without errors, allowing the amount to be returned to the customer.

### Actual behavior

The system blocks the refund process and shows the error message: Order does not have an outstanding balance to refund.

Additional Context:

- The payment is fully captured.
- The order status shows as eligible for a refund.
- No other balance-related issues are visible.

![image](https://github.com/user-attachments/assets/72709902-3f40-47d3-8f33-9b5c1b2cde6e)

### Link to reproduction repo

Cloned new with 2.0.7

**Comment by carlos-r-l-rodrigues** [comment]
Hi @usamaadev,
thanks for the post, but this is an intended validation.
In order to create a refund, the Order balance has to be below 0, meaning an item was returned, or something else edited.
We might revisit this behavior in the future, but we'll keep it as it is for now, considering it has some accounting implications that make this trickier than it seems to be.

**Comment by olivermrbl** [comment]
@usamaadev, we are currently scoping this feature, and it would be helpful to understand your use case to ensure it is addressed. Can I get you to share more about what you intend to use "custom" refunds for?

**Comment by xqSimone** [comment]
hi @olivermrbl 

Our use case is that we basically sell tickets for events, I try to sum up our cases in regards to the current refund process, I am still new to medusa so it could be that I still misunderstand something in what is currently possible or not.

case 1 - Customer buy a course seat (on premise, with limited amount), next week the price may go down, we refund the difference without a "Return" 

case 2 - Customer buy a course seat (on premise, with limited amount), ask for a refund, this is possible to handle right now but it is necessary to create fulfillment, ship, and return an item that was actually never shipped.

case 3 - Customer buys a digital product and ask for a refund, which seems not covered by the documented Digital Producs Recipe. In this case since it is a digital product there would be no fulfillment, ship, and return.

https://docs.medusajs.com/resources/recipes/digital-products

So to sum it up we would need more flexibility in issuing refunds without having to "ship" "return" "deliver" "receive" the "items".

I found a reddit question on a similar topic where it could be you replied as well.

https://www.reddit.com/r/webdev/comments/192s0jk/headless_ecom_platform_for_selling_tickets/?rdt=59111

**Comment by jgatto1** [comment]
> Hi [@usamaadev](https://github.com/usamaadev), thanks for the post, but this is an intended validation. In order to create a refund, the Order balance has to be below 0, meaning an item was returned, or something else edited. We might revisit this behavior in the future, but we'll keep it as it is for now, considering it has some accounting implications that make this trickier than it seems to be.

Thanks @carlos-r-l-rodrigues I created a return and it worked.
